lib/Jsrt/ChakraCommon.h

```diff
@@ -2978,4 +2978,343 @@ typedef unsigned char* ChakraBytePtr;
+    CHAKRA_API
+        JsTTDStopTimeTravelRecording();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Emit Time-Travel Recording.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDEmitTimeTravelRecording();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Start Time-Travel Debugging.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDStartTimeTravelDebugging();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Pause Time-Travel recording gefor executing code on behalf of debugger or other diagnostic/telemetry.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDPauseTimeTravelBeforeRuntimeOperation();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     ReStart Time-Travel recording after executing code on behalf of debugger or other diagnostic/telemetry.
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDReStartTimeTravelAfterRuntimeOperation();
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Notify the Js runtime that the host as created/canceled a callback with the given function and id.
+    /// </summary>
+    /// <param name="isCancel">True if the action is to create the callback with the callbackId.</param>
+    /// <param name="isCancel">True if the action is to cancel the callback with the callbackId.</param>
+    /// <param name="isRepeating">True if the action is to create a repeating callback (e.g., setInterval).</param>
+    /// <param name="function">The function associated with the callbackId.</param>
+    /// <param name="callbackId">The callbackId that is being created/canceled.</param>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API 
+        JsTTDNotifyHostCallbackCreatedOrCanceled(
+            _In_ bool isCreated,
+            _In_ bool isCancel, 
+            _In_ bool isRepeating, 
+            _In_ JsValueRef function, 
+            _In_ INT64 callbackId);
+
+    /// <summary>
+    ///     TTD API -- may change in future versions:
+    ///     Notify the Js runtime we are at a safe yield point in the event loop (i.e. no locals on the stack and we can proccess as desired).
+    /// </summary>
+    /// <returns>The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.</returns>
+    CHAKRA_API
+        JsTTDNotifyYield();
```

Comment:
Does this mean no JavaScript execution on stack? what happens if we call this in between?